docs/snippets/vue/page-story-slots.3.ts.mdx

```diff
@@ -5,19 +5,11 @@ import type { Meta, StoryObj } from '@storybook/vue3';
 
 import Page from './Page.vue';
 
-const meta: Meta<typeof Page> = {
-  component: Page,
-};
-
-export default meta;
-type Story = StoryObj<typeof Page>;
+// TODO: How do you extract a Vue3 component's props type?
+// type PagePropsAndCustomArgs = Page & { footer?: string };
```

Comment:
I don't think there exist something official, I do some conditional typing magic in the implementation:
```ts
type ComponentProps<C> = C extends ExtendedVue<any, any, any, any, infer P>
  ? P
  : C extends Component<any, any, any, infer P>
  ? P
  : unknown;
```